### PR TITLE
FOUNDATIONS: Retrieve Intern By Id

### LINKED ISSUE #52: FOUNDATIONS: Retrieve Intern By Id

### PR DESCRIPTION
**Pull request by ADILATIC**
Closes #52 

image

### CODE REVIEW

**Comment by ADILATIC** [comment]
image

**Comment by ADILATIC** [comment]
image

**Comment by ADILATIC** [comment]
image

**Comment by ADILATIC** [comment]
image

**Comment by ADILATIC** [comment]
image

**Comment by ADILATIC** [comment]
image

File: InternTrack.Core.Api.Tests.Unit/Services/Foundations/Interns/InternServiceTests.Exceptions.RetrieveById.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+            var serviceException = new Exception();
+
+            var failedInternServiceException =
+                new FailedInternServiceException(serviceException);
+
+            var expectedInternServiceException =
+                new InternServiceException(failedInternServiceException);
+
+            this.storageBrokerMock.Setup(broker =>
+            broker.SelectInternByIdAsync(inputInternId))
```
Comment (ElbekDeveloper) [inline InternTrack.Core.Api.Tests.Unit/Services/Foundations/Interns/InternServiceTests.Exceptions.RetrieveById.cs:32]: tab here and use `someInternId` and get rid of `inputInternId`

File: InternTrack.Core.Api.Tests.Unit/Services/Foundations/Interns/InternServiceTests.Exceptions.RetrieveById.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+using InternTrack.Core.Api.Models.Interns;
+using InternTrack.Core.Api.Models.Interns.Exceptions;
+using Moq;
+using Xunit;
+
+namespace InternTrack.Core.Api.Tests.Unit.Services.Foundations.Interns
+{
+    public partial class InternServiceTests
+    {
+        [Fact]
```
Comment (ElbekDeveloper) [inline InternTrack.Core.Api.Tests.Unit/Services/Foundations/Interns/InternServiceTests.Exceptions.RetrieveById.cs:17]: You need to also handle sql exception (i.e. `ShouldThrowCriticalDependencyExceptionOnRetrieveAllWhenSqlExceptionOccursAndLogIt`);

**Review by ElbekDeveloper — CHANGES_REQUESTED** [review]

File: InternTrack.Core.Api.Tests.Unit/Services/Foundations/Interns/InternServiceTests.Exceptions.RetrieveById.cs [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+                broker.SelectInternByIdAsync(It.IsAny<Guid>()))
+                    .ThrowsAsync(sqlException);
+
+            // when
+            ValueTask<Intern> retrieveInternByIdTask =
+                this.internService.RetrieveInternByIdAsync(someInternId);
+
+            // then
+            await Assert.ThrowsAsync<InternDependencyException>(() =>
+                retrieveInternByIdTask.AsTask());
```
Comment (ElbekDeveloper) [inline InternTrack.Core.Api.Tests.Unit/Services/Foundations/Interns/InternServiceTests.Exceptions.RetrieveById.cs:40]: This is the old way of checking exceptions

File: InternTrack.Core.Api.Tests.Unit/Services/Foundations/Interns/InternServiceTests.Exceptions.RetrieveById.cs [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+                broker.SelectInternByIdAsync(someInternId))
+                    .ThrowsAsync(serviceException);
+
+            // when
+            ValueTask<Intern> retrieveInternByIdTask =
+                this.internService.RetrieveInternByIdAsync(someInternId);
+
+            // then
+            await Assert.ThrowsAsync<InternServiceException>(() =>
+                retrieveInternByIdTask.AsTask());
```
Comment (ElbekDeveloper) [inline InternTrack.Core.Api.Tests.Unit/Services/Foundations/Interns/InternServiceTests.Exceptions.RetrieveById.cs:80]: Change this to newer way of asserting exceptions.

File: InternTrack.Core.Api.Tests.Unit/Services/Foundations/Interns/InternServiceTests.Validations.RetrieveById.cs [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+            var expectedInternValidationException =
+                new InternValidationException(invalidInternException);
+
+            // when
+            ValueTask<Intern> actualInternTask =
+                this.internService.RetrieveInternByIdAsync(invalidInternId);
+
+            // then
+            await Assert.ThrowsAsync<InternValidationException>(() => actualInternTask.AsTask());
```
Comment (ElbekDeveloper) [inline InternTrack.Core.Api.Tests.Unit/Services/Foundations/Interns/InternServiceTests.Validations.RetrieveById.cs:36]: upgrade to newer way

File: InternTrack.Core.Api.Tests.Unit/Services/Foundations/Interns/InternServiceTests.Validations.RetrieveById.cs [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogError(It.Is(SameValidationExceptionAs(
+                    expectedInternValidationException))),
+                        Times.Once);
+
+            this.storageBrokerMock.Verify(broker =>
+                broker.SelectInternByIdAsync(It.IsAny<Guid>()),
+                    Times.Never);
+
```
Comment (ElbekDeveloper) [inline InternTrack.Core.Api.Tests.Unit/Services/Foundations/Interns/InternServiceTests.Validations.RetrieveById.cs:46]: This should come after logging broker here

**Review by ElbekDeveloper — CHANGES_REQUESTED** [review]

File: InternTrack.Core.Api.Tests.Unit/Services/Foundations/Interns/InternServiceTests.Exceptions.RetrieveById.cs (hunk 3 above)
Comment (glhays) [reply InternTrack.Core.Api.Tests.Unit/Services/Foundations/Interns/InternServiceTests.Exceptions.RetrieveById.cs:40]: @ADILATIC here is a [reference.](Tarteeb.Api.Tests.Unit/Services/Foundations/Tickets/TicketServiceTests.Exceptions.RetrievebyId.cs) You have to look for newer projects within the community. The Standard is ever changing and being improved upon.

**Comment by ADILATIC** [comment]
image

**Comment by ADILATIC** [comment]
I resolved the issues from my reviews, here is the final screen shot of the passing tests.
image

**Pull request closed without merging**